docs/src/main/asciidoc/dev-services.adoc

```diff
@@ -0,0 +1,75 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Dev Services Overview
+:extension-status: preview
+
+include::./attributes.adoc[]
+
+Quarkus supports the automatic provisioning of unconfigured services in development and test mode. We refer to this capability
+as Dev Services. From a developers perspective this means that if you include an extension and don't configure it then
+Quarkus will automatically start the relevant service (usually using Test Containers behind the scenes) and wire up your
+application to use this service.
+
+All this functionality is part of the Quarkus `deployment` modules, so does not affect the production application in any
+way.
+
+This page lists all the Dev Services that Quarkus supports.
+
+NOTE: In order to use Dev Services you will generally need a working docker environment (remote environments are supported).
+If you don't have docker installed you will need to configure your services normally.
+
+== Amqp
+
+Amqp will be enabled by adding `quarkus-smallrye-reactive-messaging-amqp` to your application.
+
+include::{generated-dir}/config/quarkus-smallrye-reactivemessaging-amqp-config-group-amqp-dev-services-build-time-config.adoc[opts=optional, leveloffset=+1]
+
+== Apicurio Registry
+
+Apicurio will be enabled by adding `quarkus-apicurio-registry-avro` to your application.
+
+include::{generated-dir}/config/quarkus-apicurio-registry-devservices-apicurio-registry-avro-apicurio-registry-dev-services-build-time-config.adoc[opts=optional, leveloffset=+1]
+
+== Databases
+
+These will be enabled by including a reactive or JDBC datasource extension.
+
+Quarkus provides Dev Services for all databases it supports except Oracle. Most of these are run in a container, with the
+exception of H2 and Derby which are run in process. Dev Services are supported for both JDBC and reactive drivers.
+
+include::{generated-dir}/config/quarkus-datasource-config-group-dev-services-build-time-config.adoc[opts=optional, leveloffset=+1]
+
+== Kafka
+
+include::{generated-dir}/config/quarkus-kafka-client-config-group-kafka-dev-services-build-time-config.adoc[opts=optional, leveloffset=+1]
+
+== Keycloak
+
+Keycloak will be enabled by including `quarkus-oidc` to your application.
+
+See link:security-openid-connect-dev-services[OIDC Dev Services]
+
+// currently its not possible to include two different config items with Duration
+// see https://github.com/quarkusio/quarkus/issues/19330
+// include::{generated-dir}/config/quarkus-keycloak-devservices-keycloak-keycloak-build-time-config.adoc[opts=optional, leveloffset=+1]
+
+== Mongo DB
+
+Mongo will be enabled by including `quarkus-mongodb-client` to your application.
+
+include::{generated-dir}/config/quarkus-mongodb-config-group-dev-services-build-time-config.adoc[opts=optional, leveloffset=+1]
```

Comment:
Not much info here: https://quarkus.io/guides/mongodb#dev-services but we should probably add the config properties there too.